docs/sources/operators-guide/deploying-grafana-mimir/jsonnet/memberlist.md

```diff
@@ -0,0 +1,147 @@
+Applying this change will not cause restart of components.
+
+You can monitor following metrics to check if mirroring was enabled on all components and if it works correctly:
+
+- `cortex_multikv_mirror_enabled`
+- `rate(cortex_multikv_mirror_writes_total[1m])`
+- `rate(cortex_multikv_mirror_write_errors_total[1m])`
+
+After mirroring is enabled Memberlist admin page should display following keys in the KV store for all components using hash ring:
+
+- `compactor` (compactors ring)
+- `distributor` (distributors ring)
+- `multi-zone/store-gateway` (or just `store-gateway` if multi-zone store gateways are not enabled).
+- `ring` (ingesters ring)
+- `rulers/ring` (if ruler is running)
+- `alertmanager` (if alertmanager is running)
+
+## Step 3: Switch Primary and Secondary store
+
+```jsonnet
+{
+  _config+:: {
+    memberlist_ring_enabled: true,
+    multikv_migration_enabled: true,
+    multikv_mirror_enabled: true,
+    multikv_switch_primary_secondary: true,  // Changed in this step.
+  }
+}
+```
+
+This change will switch primary and secondary stores as used by `multi` KV.
+From this point on components will use memberlist as primary KV store, and they will mirror updates to Consul.
+This step does NOT require restart of components.
+
+To see if all components started to use `memberlist` as primary store, please watch `cortex_multikv_primary_store` metric.
+
+## Step 4: Disable mirroring to Consul
+
+```jsonnet
+{
+  _config+:: {
+    memberlist_ring_enabled: true,
+    multikv_migration_enabled: true,
+    multikv_mirror_enabled: false,  // Changed in this step.
+    multikv_switch_primary_secondary: true,
+  }
+}
+```
+
+This step doesn't require restart. After applying the change components will stop writing ring updates to Consul, and will only use `memberlist`.
+You can watch `cortex_multikv_mirror_enabled` metric to see if all components have picked up updated configuration.
+
+## Step 5: Disable `multi` KV Store
+
+```jsonnet
+{
+  _config+:: {
+    memberlist_ring_enabled: true,
+    multikv_migration_enabled: false,  // Changed in this step.
+    multikv_mirror_enabled: false,
+    multikv_switch_primary_secondary: true,
+    multikv_migration_teardown: true,  // Added in this step.
+  }
+}
+```
+
+This configuration change will cause a new rollout of all components.
+After restart components will no longer use `multi` KV store and will be configured to use `memberlist` only.
+We use `multikv_migration_teardown` to preserve runtime configuration for `multi` KV store for components that haven't restarted yet.
+
+All `cortex_multikv_*` metrics are only exposed by components that use `multi` KV store. As components restart, these metrics will disappear.
+
+**Note: setting `multikv_migration_enabled: false` while keeping `memberlist_ring_enabled: true` will also remove Consul! That's expected, since Consul is not used anymore – mirroring to it was disabled in step 4.**
```

Comment:
Per docs squad guidelines, syntax for notes is:

```suggestion
> **Note**: setting `multikv_migration_enabled: false` while keeping `memberlist_ring_enabled: true` will also remove Consul! That's expected, since Consul is not used anymore – mirroring to it was disabled in step 4.
```